### PR TITLE
TEST: Randomize soft-deletes settings

### PR DESCRIPTION
**Pull request by dnhatn**
This change allows us to test our system with/without soft-deletes enabled.

### CODE REVIEW

**Review by bleskes — APPROVED** [review]
LGTM as long as CI is happy. Also please run for a good day on your CI before merging. I don't think one successful CI run is enough for this one.

**Comment by dnhatn** [comment]
> testAuthenticateCaching failed.
@elasticmachine test this please.

**Comment by dnhatn** [comment]
It passed several iterations on my CI. I will merge this and watch the CCR builds. Thanks @bleskes.

**Pull request merged**
